internal/deploys/handler_test.go

```diff
@@ -0,0 +1,138 @@
+package deploys
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/broadinstitute/sherlock/internal/builds"
+	"github.com/broadinstitute/sherlock/internal/testutils"
+	"github.com/bxcodec/faker/v3"
+)
+
+func (suite *DeployIntegrationTestSuite) TestCreateDeployHandler() {
+	// setup prepopulate a service instance and some builds
+	var preExistingServiceInstanceReq CreateServiceInstanceRequest
+	err := faker.FakeData(&preExistingServiceInstanceReq)
+	suite.Require().NoError(err)
+	preExistingServiceInstance, err := suite.app.deploys.serviceInstances.CreateNew(preExistingServiceInstanceReq)
+	suite.Require().NoError(err)
+
+	// make a build associated with the service from the service instance above
+	var preExistingBuildFromServiceReq builds.CreateBuildRequest
+	err = faker.FakeData(&preExistingBuildFromServiceReq)
+	suite.Require().NoError(err)
+	preExistingBuildFromServiceReq.ServiceName = preExistingServiceInstance.Service.Name
+	preExistingBuildFromService, err := suite.app.deploys.builds.CreateNew(preExistingBuildFromServiceReq)
+	suite.Require().NoError(err)
+
+	// make a build unassociated with any service instance
+	var otherPreExistingBuildReq builds.CreateBuildRequest
+	err = faker.FakeData(&otherPreExistingBuildReq)
+	suite.Require().NoError(err)
+	otherPreExistingBuild, err := suite.app.deploys.builds.CreateNew(otherPreExistingBuildReq)
+	suite.Require().NoError(err)
+
+	suite.Run("creates deploy from existing service instance and build", func() {
+		ctx, response := testutils.SetupTestContext()
+		deployReq := CreateDeployRequestBody{
+			VersionString: preExistingBuildFromService.VersionString,
+		}
+
+		reqBody := new(bytes.Buffer)
+		err := json.NewEncoder(reqBody).Encode(deployReq)
+		suite.Require().NoError(err)
+		path := fmt.Sprintf("/deploys/%s/%s", preExistingServiceInstance.Environment.Name, preExistingServiceInstance.Service.Name)
+		req, err := http.NewRequest(http.MethodPost, path, reqBody)
+		suite.Require().NoError(err)
+		ctx.Request = req
+
+		suite.app.deploys.createDeploy(ctx)
+		suite.Assert().Equal(http.StatusCreated, response.Code)
```

Comment:
I think we return the serialized deploy here so we should confirm it's correctly updated and not half-formed somehow.